lib/generic/cluster_descriptor.h

```diff
@@ -7,18 +7,18 @@
 
 namespace MouseTrack {
 
-
-/// This is an abstract class. Derive from this class to create a usable cluster descriptor.
+/// This is an abstract class. Derive from this class to create a usable cluster
+/// descriptor.
 class ClusterDescriptor {
 public:
-	/// Returns a distance/norm `n` between `this` and `other`.
-	/// It should be small for two instances that are "similar" to each other, 
-	/// and large for two instances that are very different.
-	/// Important properties:
-	/// 1. 0 <= n <= inf
-	/// 2. if this == other -> n == 0
-	virtual double compare(const ClusterDescriptor* other) const = 0;
+  virtual ~ClusterDescriptor(){};
```

Comment:
Thanks for adding :+1: 
Can you move the implementation to the cpp? Moving as much as possible into the cpp saves compile time.

Reply:
You would really prefer adding a whole .cpp file just to implement an empty destructor?